lib/adagrams.rb

```diff
@@ -0,0 +1,90 @@
+require "awesome_print"
+
+def draw_letters
+  avail_letters = %w[A A A A A A A A A B B C C D D D D E E E E E E E E E E E E F F G G G H H I I I I I I I I I J K L L L L M M N N N N N N O O O O O O O O P P Q R R R R R R S S S S T T T T T T U U U U V V W W X Y Y Z]
+  avail_letters.sample(10)
+end
+
+sample = draw_letters
+# print sample
+
+def uses_available_letters?(input, letters_in_hand)
+  letters_in_hand = letters_in_hand.clone
+  input = input.upcase
+  input.each_char.all? do |char|
+    if letters_in_hand.include?(char)
+      letters_in_hand.delete_at(letters_in_hand.index(char))
+      true
+    else
+      false
+    end
+  end
+end
+
+def score_word(word)
+  score = []
+  score_chart = [%w[A E I O U L N R S T], %w[D G], %w[B C M P], %w[F H V W Y], %w[K]]
+  word = word.upcase
+
+  score_chart.each_with_index do |letter_bank, letter_value|
+    word.each_char do |char|
+      if letter_bank.include?(char)
+        score << (letter_value + 1)
+      end
+    end
+  end
+
+  word.each_char do |char|
+    if ["J", "X"].include?(char)
+      score << 8
+    elsif ["Q", "Z"].include?(char)
+      score << 10
+    end
+  end
+
+  if word.length > 6
+    score << 8
+  end
+
+  return score.sum
```

Comment:
I think it's really fascinating that you all did an array of scores and then summed it in the end! I think most people would have opted to just add them, but I think this is also valid and interesting and cool and is very compatible with thinking like a programmer :D